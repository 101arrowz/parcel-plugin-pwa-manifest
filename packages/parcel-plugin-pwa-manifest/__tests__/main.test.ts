/// <reference types="../types" />

import Bundler from 'parcel-bundler';
import { resolve } from 'path';
import attachManifestGenerator from '../index';
import { readdirSync, readFileSync } from 'fs';
import { EventEmitter } from 'events';
import ParcelBundler from 'parcel-bundler';
const outDir = resolve(__dirname, 'tmp', Date.now().toString(36));
const bundler = new Bundler(resolve(__dirname, 'sample', 'index.html'), {
  outDir,
  contentHash: true,
  watch: false
}) as ParcelBundler & EventEmitter;
attachManifestGenerator(bundler as FullBundler);
jest.setTimeout(10000);
test('Integrated correctly', async done => {
  new Promise(res =>
    bundler.on('buildEnd', () => {
      expect(readdirSync(outDir)).toMatchSnapshot();
      expect(
        readFileSync(resolve(outDir, 'index.html')).toString()
      ).toMatchSnapshot();
      expect(
        readFileSync(resolve(outDir, 'browserconfig.xml')).toString()
      ).toMatchSnapshot();
      expect(
        JSON.parse(
          readFileSync(resolve(outDir, 'manifest.webmanifest')).toString()
        )
      ).toMatchSnapshot();
      res();
    })
  ).then(done);
});
bundler.bundle();
