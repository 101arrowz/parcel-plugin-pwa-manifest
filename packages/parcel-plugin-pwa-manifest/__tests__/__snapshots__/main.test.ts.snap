// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integrated correctly 1`] = `
Array [
  "apple-touch-icon.9626d75a.png",
  "browserconfig.xml",
  "icon-192x192.1ca0d54e.png",
  "icon-192x192.6cf01f66.webp",
  "icon-512x512.9de13230.webp",
  "icon-512x512.e481a035.png",
  "index.html",
  "manifest.webmanifest",
  "mstile-150x150.e286440d.png",
  "mstile-310x150.926bd872.png",
  "mstile-310x310.e10f7557.png",
  "mstile-70x70.3af68a9a.png",
]
`;

exports[`Integrated correctly 2`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"UTF-8\\">
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
  <title>My PWA</title>
<meta name=\\"msapplication-config\\" content=\\"/browserconfig.xml\\"><meta name=\\"theme-color\\" content=\\"white\\"><link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"/apple-touch-icon.9626d75a.png\\"><link rel=\\"manifest\\" href=\\"/manifest.webmanifest\\"></head>
<body>
</body>
</html>"
`;

exports[`Integrated correctly 3`] = `"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?><browserconfig><msapplication><tile><TileColor>white</TileColor><square70x70logo src=\\"/mstile-70x70.3af68a9a.png\\"/><square150x150logo src=\\"/mstile-150x150.e286440d.png\\"/><square310x310logo src=\\"/mstile-310x310.e10f7557.png\\"/><wide310x150logo src=\\"/mstile-310x150.926bd872.png\\"/></tile></msapplication></browserconfig>"`;

exports[`Integrated correctly 4`] = `
Object {
  "background_color": "white",
  "description": "A PWA",
  "display": "standalone",
  "icons": Array [
    Object {
      "sizes": "192x192",
      "src": "/icon-192x192.6cf01f66.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "192x192",
      "src": "/icon-192x192.1ca0d54e.png",
      "type": "image/png",
    },
    Object {
      "sizes": "512x512",
      "src": "/icon-512x512.9de13230.webp",
      "type": "image/webp",
    },
    Object {
      "sizes": "512x512",
      "src": "/icon-512x512.e481a035.png",
      "type": "image/png",
    },
  ],
  "name": "my-pwa",
  "scope": "/",
  "short_name": "my-pwa",
  "start_url": "/",
  "theme_color": "white",
}
`;
